src/controllers/eventController.js

```diff
@@ -15,9 +15,15 @@ const { checkRequestData } = require('../utils/checkData');
 
 eventController.post('/register', async (req, res) => {
     try {
+        if (req.requester.isApproved === false) {
+            throw new Error('This "User" is not approved to register an Event!')
+        }
         if (Object.keys(req.body).length === 0) {
             throw new Error('Invalid request Body!')
         }
+        if (req.body.dates[0].startTime >= req.body.dates[0].endTime) {
```

Comment:
Why checking only the first date and is it tested with different scenarios?

Reply:
Yeah forgot about this, this was only for testing. Thanks, i have made a function about duplicate dates now.